src/data/finsupp/basic.lean

```diff
@@ -774,8 +775,97 @@ end
 @[simp] lemma graph_eq_empty {f : α →₀ M} : f.graph = ∅ ↔ f = 0 :=
 (graph_injective α M).eq_iff' graph_zero
 
+/-- When turned into a list of key/value pairs, a graph has no duplicate keys. -/
+lemma graph_nodupkeys (f : α →₀ M) : (list.map prod.to_sigma f.graph.to_list).nodupkeys :=
```

Comment:
I see `map prod.sigma f.graph_to_list` repeated a lot, should that get its own definition `f.to_alist`?

Reply:
Sure. I'll do that in a small, separate PR.

Reply:
Nevermind, might as well do it here.